modules/configuration/src/configuration.ts

```diff
@@ -62,8 +72,19 @@ function resolveSpecs(config: Config) {
             })
         }
     }
+    return Promise.all(promises)
+}
+function getSpecFromUrl(url: string) {
+    return new Promise((resolve, reject) => {
+        request.get({
```

Comment:
Overall I think you should use a callback approach for now to not have the "async" on the API interface.
Or really switch to it but without using an await in the code, you should be able to chain all the tasks without blocking.
Then you also need to change the API to be async and accept in the modules a Promise<Config>

Reply:
as per our conversation, we will leave as is